File: 0347_Top_K_Frequent_Elements/347.py
class Solution:
    def topKFrequent(self, nums, k):
        d = {}
        ans = []
        for i in nums:
            if i in d.keys():
                d[i] += 1
            else:
                d[i] = 1
        # print(d)
        L = list(d.items())
        # print(L)
        L.sort(key = lambda tup: tup[1], reverse = True)
        for i in range(k):
            ans.append(L[i][0])
        return ans

a = Solution()
print(a.topKFrequent(nums = [1,1,1,2,2,3], k = 2))
print(a.topKFrequent(nums = [1], k = 1))

File: 0347_Top_K_Frequent_Elements/347.cpp
#include <iostream>
#include <map>
#include <queue>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> times;
        priority_queue<pair<int, int>> pq;
        vector<int> ans;
        for (int i: nums)
            times[i]++;
        for (map<int, int>::iterator it = times.begin(); it != times.end(); ++it) {
            // cout << it->first << " " << it->second << "\n";
            pq.push({it->second, it->first});
        }
        for (int _ = 0; _ < k; ++_) {
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};

int main() {
    Solution a;
    vector<int> n = {1, 1, 1, 2, 2, 3};
    for (int num: a.topKFrequent(n, 2)) 
        cout << num << " ";
    cout << "\n";
    return 0;
}